### PR TITLE
CSI 0.2.0 breaking changes

### PR DESCRIPTION
**Pull request by sbezverk**
WIP dp not merge until official CSI 0.2.0 spec is out

### CODE REVIEW

File: driver/driver.go [hunk 1]
```diff
@@ -14,16 +14,14 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-//go:generate mockgen -package=driver -destination=driver.mock.go github.com/container-storage-interface/spec/lib/go/csi IdentityServer,ControllerServer,NodeServer
```
Comment (lpabon) [inline driver/driver.go]: I think this is an old version of this file. I remember fixing a lot of this.

Comment (sbezverk) [reply driver/driver.go]: @lpabon not sure what to say, it was in csi-test master.

Comment (lpabon) [reply driver/driver.go]: Looks like you may want to rebase your branch. Check out master:
https://github.com/kubernetes-csi/csi-test/blob/f56f43f4866e6bdef38e74e69c047b8377e0221c/driver/driver.go#L17

**Comment by sbezverk** [comment]
@lpabon Hi Luis, all tests are green but I had to use a workaround since we have no control over gocsi repo. I was wondering if we could merge it into 0.2.0 branch in order to not break existing 0.1.0 users. Appreciate your thoughts here.

**Comment by sbezverk** [comment]
@lpabon we got officially 0.2.0 changes to gocsi merged into branch csi-0.2.0!!! So we are good to go here, no more hacky workaround.

File: driver/driver.mock.go [hunk 2]
```diff
@@ -1,289 +1,656 @@
-// Code generated by MockGen. DO NOT EDIT.
-// Source: github.com/container-storage-interface/spec/lib/go/csi (interfaces: IdentityServer,ControllerServer,NodeServer)
```
Comment (lpabon) [inline driver/driver.mock.go]: Please re-generate this once driver/driver.go has been replaced.

File: hack/e2e.sh [hunk 3]
```diff
@@ -23,7 +24,16 @@ runTest()
...
 }
 
-go get -u github.com/thecodeteam/gocsi/mock
+#
+# TODO Once 0.2.0 change gets merged into gocsi repo, switch back to "go get"
+#
+git clone https://github.com/thecodeteam/gocsi $GOPATH/src/github.com/thecodeteam/gocsi
+pushd $GOPATH/src/github.com/thecodeteam/gocsi
+git checkout $CSI_MOCK_VERSION
+make build
```
Comment (lpabon) [inline hack/e2e.sh]: Does this copy it to `$GOPATH/bin`?

File: test/driver_test.go [hunk 4]
```diff
@@ -40,7 +40,7 @@ func (s *simpleDriver) GetSupportedVersions(
 	context.Context, *csi.GetSupportedVersionsRequest) (*csi.GetSupportedVersionsResponse, error) {
 	return &csi.GetSupportedVersionsResponse{
 		SupportedVersions: []*csi.Version{
-			&csi.Version{
+			{
 				Major: 0,
 				Minor: 1,
```
Comment (lpabon) [inline test/driver_test.go]: Probably need to change this?

File: driver/driver.mock.go (hunk 2 above)
Comment (sbezverk) [reply driver/driver.mock.go]: Please let me know steps to re-generate it.

File: hack/e2e.sh (hunk 3 above)
Comment (sbezverk) [reply hack/e2e.sh]: Yes, otherwise e2e test would have filed.

File: test/driver_test.go (hunk 4 above)
Comment (sbezverk) [reply test/driver_test.go]: will do.

**Pull request merged**